process_request/helpers.py

```diff
@@ -39,12 +37,34 @@ def check_for_instance_type(archival_object, type_to_check):
 
 def get_dates(archival_object):
     """Takes json for an archival object that has the _resolved parameter on ancestors. Gets date expression for an item. Starts at item level, goes up until a date is found"""
+    dates = []
     if archival_object.get("dates"):
-        return archival_object.get("dates")[0].get("expression")
+        for d in archival_object.get("dates"):
+            dates.append(get_expression(d))
```

Comment:
Again, an opportunity for list comprehension (which you can always do if you're just performing one action inside a for loop:
```suggestion
        dates = [get_expression(d) for d in archival_object.get("dates")]
```